jiant/tasks/seq2seq.py

```diff
@@ -71,7 +72,10 @@ def get_data_iter(self, path):
                 if len(row) < 2 or not row[0] or not row[1]:
                     continue
                 src_sent = tokenize_and_truncate(self._tokenizer_name, row[0], self.max_seq_len)
-                tgt_sent = tokenize_and_truncate(self._tokenizer_name, row[2], self.max_seq_len)
+                if self.val_metric == "%s_accuracy" % self.name:
+                    tgt_sent = tokenize_and_truncate(self._tokenizer_name, row[2], self.max_seq_len)
+                else:  # for MT
```

Comment:
This looks dangerous, especially since val_metric is user-settable. Find another way to distinguish the two task types.

Reply:
@pruksmhc Could you please have a look at this, too? What would be the best way to distinguish between tasks? (I think Sam was against using task names.)

Reply:
Honestly, what I would do is create another class that inherits from Seq2SeqTask (MTTask for example), which uses BLEU. Then, the flow for everything in that class would be for BLEU, and avoids too many if-elses in the code.